src/containers/Profiles/Profiles.tsx

```diff
@@ -0,0 +1,84 @@
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router';
+import shallow from 'zustand/shallow';
+import { useTranslation } from 'react-i18next';
+
+import styles from './Profiles.module.scss';
+
+import ProfileBox from '#src/components/ProfileBox/ProfileBox';
+import { useAccountStore } from '#src/stores/AccountStore';
+import type { Profile } from '#types/account';
+import AddNewProfile from '#src/components/ProfileBox/AddNewProfile';
+import LoadingOverlay from '#src/components/LoadingOverlay/LoadingOverlay';
+import Button from '#src/components/Button/Button';
+import { useSelectProfile, useListProfiles, useProfilesFeatureEnabled } from '#src/hooks/useProfiles';
+import useBreakpoint, { Breakpoint } from '#src/hooks/useBreakpoint';
+const MAX_PROFILES = 4;
+
+type Props = {
+  editMode?: boolean;
+};
+
+const Profiles = ({ editMode = false }: Props) => {
+  const navigate = useNavigate();
+  const { t } = useTranslation('user');
+  const { loading, user } = useAccountStore(({ loading, user }) => ({ loading, user }), shallow);
+  const profilesEnabled = useProfilesFeatureEnabled();
+
+  const breakpoint: Breakpoint = useBreakpoint();
+  const isMobile = breakpoint === Breakpoint.xs;
+
+  const { data, isLoading, isFetching, refetch, isError } = useListProfiles();
+  const activeProfiles = data?.responseData.collection.length || 0;
+  const canAddNew = activeProfiles < MAX_PROFILES;
+
+  useEffect(() => {
+    if (!profilesEnabled || !user?.id || isError) navigate('/');
```

Comment:
Will this not create a redirect loop because the index route redirects back to the profiles page?